src/OpenTelemetry/Resources/Resource.cs

```diff
@@ -35,7 +35,7 @@ public class Resource
         /// Initializes a new instance of the <see cref="Resource"/> class.
         /// </summary>
         /// <param name="attributes">An <see cref="IEnumerable{T}"/> of attributes that describe the resource.</param>
-        internal Resource(IEnumerable<KeyValuePair<string, object>> attributes)
+        public Resource(IEnumerable<KeyValuePair<string, object>> attributes)
```

Comment:
please keep this internal as before.
We already exposed IResourceDetector, and we are missing a public method to add detectors : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Resources/ResourceBuilder.cs#L89-L91 --> this is the thing to public ize to address the issue.

Reply:
After re-reading, and offline discussion, we'll have to expose this public, as Detectors need to return Resource, so need public ctor to do that.